crates/autopilot/src/database/onchain_order_events.rs

```diff
@@ -419,6 +423,32 @@ fn extract_invalidated_order_uids(
         .collect()
 }
 
+/// Temporarily stores metrics for onchain order parsing, which will later be written to the global
+/// metrics object.
+struct MetricsOnchainOrderParsing {
+    errors: Arc<RwLock<HashMap<String, u64>>>,
+}
+impl MetricsOnchainOrderParsing {
+    fn new() -> Self {
+        Self {
+            errors: Arc::new(RwLock::new(HashMap::<String, u64>::new())),
+        }
+    }
+    fn write_to_metrics(self, metrics: &'static Metrics) {
+        let MetricsOnchainOrderParsing { errors } = self;
+        if let Ok(read_lock) = errors.read() {
+            for (error_label, increment) in read_lock.iter() {
+                metrics
+                    .onchain_order_errors
+                    .with_label_values(&[error_label])
+                    .inc_by(*increment);
```

Comment:
I believe if all you do is increasing a labeled counter there is probably no advantage to having the intermediate `MetricsOnchainOrderParsing` instance because there is probably no harm in having partially updated metrics.
And even with your intermediate struct the update to the metrics will not be atomic anyway.

Reply:
I removed the intermediate struct.